SpokeStack/TFLiteWakewordRecognizer.swift

```diff
@@ -0,0 +1,478 @@
+                    assert(model.inputTensorCount == Tensors.allCases.count)
+                } else {
+                    throw WakewordModelError.encode("encode.lite was not initialized")
+                }
+                
+                self.detectModel = try Interpreter(modelPath: c.detectModelPath)
+                if let model = self.detectModel {
+                    try model.allocateTensors()
+                } else {
+                    throw WakewordModelError.detect("detect.lite was not initialized")
+                }
+            } catch let message {
+                assertionFailure("TFLiteWakewordRecognizer setConfiguration \(message)")
+            }
+        }
+    }
+    
+    private func setConfiguration() -> Void {
+        if let c = self.configuration {
+
+            /// VAD configuration
+            do {
+                try self.vad.create(mode: .HighQuality, delegate: self, frameWidth: c.frameWidth, samplerate: c.sampleRate)
+            } catch {
+                assertionFailure("TFLiteWakewordRecognizer failed to create a valid VAD")
+            }
+            
+            /// Signal normalization
+            self.rmsAlpha = c.rmsAlpha
+            self.rmsTarget = c.rmsTarget
+            self.rmsValue = self.rmsTarget
+            self.preEmphasis = c.preEmphasis
+
+            /// Sliding window buffers
+            self.fftFrame = Array(repeating: 0.0, count: c.fftWindowSize)
+            self.melWidth = c.melFrameWidth
+            self.hopLength = c.fftHopLength * c.sampleRate / 1000
+            let melLength: Int = c.melFrameLength * c.sampleRate / 1000 / self.hopLength
+            self.frameWindow = RingBuffer(melLength * self.melWidth, repeating: 0.0)
+            self.sampleWindow = RingBuffer(c.fftWindowSize, repeating: 0.0)
+            self.fftWindow = SignalProcessing.hannWindow(c.fftWindowSize)
+            self.fft = FFT(c.fftWindowSize)
+            
+            /// Attention model buffers
+            self.encodeWidth = c.encodeWidth
+            self.encodeLength = c.encodeLength * c.sampleRate / 1000 / self.hopLength
+            self.stateWidth = c.stateWidth
+            self.encodeWindow = RingBuffer(self.encodeLength * c.encodeWidth, repeating: 0.0)
+            self.encodeState = RingBuffer(c.stateWidth, repeating: 0.0)
+            self.encodeState.fill(0.0)
+            self.detectWindow = RingBuffer(self.encodeLength * c.encodeWidth, repeating: 0.0)
+            
+            /// attention model posteriors
+            self.posteriorThreshold = c.wakeThreshold
+            self.posteriorMax = 0
+            
+            /// Calculate the wakeword activation lengths
+            let frameWidth: Int = c.frameWidth
+            self.minActive = c.wakeActiveMin / frameWidth
+            self.maxActive = c.wakeActiveMax / frameWidth
+        }
+    }
+    
+    /// MARK: Pipeline control
+    
+    private func process(_ frame: Data) -> Void {
+        self.activeLength += 1
```

Comment:
The logic here is incorrect. The active length increment and max active check should only be done if the pipeline is active: https://github.com/pylon/spokestack-android/blob/master/src/main/java/com/pylon/spokestack/wakeword/WakewordTrigger.java#L357.

It's dangerous, because every `activeLength` frames, you are going to fall into the else side and reset all of the sliding windows, even if the pipeline is not active.